task6/js/script.js

```diff
@@ -0,0 +1,521 @@
+   validUntil: new Date('2021-05-10T23:00:00'),
+   discount: '40%',
+   hashTags: ['одежда'],
+   rating: 4.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '16',
+   description: 'Скидка на носки -12%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://oz.by/kidssocks/more10979463.html?sbtoken=215530f181b9aa0b4377db923b2774ea',
+   vendor: 'OZ',
+   photoLink: 'https://s2-goods.ozstatic.by/2000/463/979/10/10979463_0.jpg',
+   validUntil: new Date('2021-03-30T23:00:00'),
+   discount: '12%',
+   hashTags: ['одежда'],
+   rating: 0.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '17',
+   description: 'Скидка на лак для волос -50%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://mila.by/catalog/ukhod-za-soboy/ukhod-za-volosami/sredstva-dlya-ukladki-volos-i-sokhraneniya-pricheski/laki/laki/lak-d-volos-sverkhsilnoy-fiksatsi-taft-ultra-neytralnyy-zapakh-225ml.html',
+   vendor: 'Мила',
+   photoLink: 'https://mila.by/upload/iblock/d21/d212e836251884a634764ad1afd41270.jpg',
+   validUntil: new Date('2021-04-15T23:00:00'),
+   discount: '50%',
+   hashTags: ['уход'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '18',
+   description: 'Скидка на кошелек -23%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://makey-shop.by/katalog/koshelki/muzhskie/koshelek-042-08-47.html',
+   vendor: 'Макей',
+   photoLink: 'https://makey-shop.by/assets/images/products/171/17027782795c2dd73fc9325-original1.jpg',
+   validUntil: new Date('2021-05-04T23:00:00'),
+   discount: '23%',
+   hashTags: ['кошелек'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '19',
+   description: 'Скидка на женские очки -34%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://funoptik.by/ochki/jenskie-ochki/jenskie-ochki-dacchi-d37412-c5',
+   vendor: 'Fun Optik',
+   photoLink: 'https://funoptik.by/uploads/product/1600/1661/thumbs/70_Dacchi-D-37412-C5.jpg',
+   validUntil: new Date('2021-06-12T23:00:00'),
+   discount: '34%',
+   hashTags: ['очки'],
+   rating: 0.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '20',
+   description: 'Скидка на маску для лица -5%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://belita-shop.by/katalog/fabric-mask/maska-dlya-litsa-na-netkanoy-osnove-sebobalans-i-uvlazhnenie.html',
+   vendor: 'Belita',
+   photoLink: 'https://belita-shop.by/upload/cacheResize/205/8a7/3b7b52476114818c686353bc18462300.jpg',
+   validUntil: new Date('2021-03-25T23:00:00'),
+   discount: '5%',
+   hashTags: ['уход'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ ]
+ let getAds = (function(skip = 0, top = 10, filterConfig = {}){
+    let adsToShow = [];
+    let key = Object.keys(filterConfig);
+
+    if(key.length == 0){ 
+        for(let i=skip;i<top+skip;++i){
+            if(i > adList.length){
+                break;
+            }
+            adsToShow.push(adList[i]);
+        }
+    } else if(key[0] == 'vendors'){
```

Comment:
Why are you checking only the first key? You can loop through all the keys using forEach